### PR TITLE
Add unit tests for StudentManager

### LINKED ISSUE #6: Add Unit Tests for StudentManager Class
The project has no tests! We need unit tests for the StudentManager class.

**Task:**
- Create a test file in the tests directory
- Write tests for at least 5 methods in StudentManager
- Use Python's unittest or pytest framework
- Ensure all tests pass

**Test Examples:**
- Test adding an assignment
- Test marking assignment as completed
- Test GPA calculation
- Test invalid grade handling

### PR DESCRIPTION
**Pull request by Rathore-tech-code**
### Overview
This PR adds unit tests for the StudentManager class.

### What was added
✔ Tests for adding assignments  
✔ Tests for marking assignments as completed  
✔ GPA calculation validation  
✔ Invalid grade handling (raises ValueError)  
✔ Student statistics calculation

### Result
All tests pass successfully using `python -m pytest`.

Fixes #6

Please review and merge 😊

### CODE REVIEW

**Review by swaparup36 — CHANGES_REQUESTED** [review]
Consider grouping related tests in classes and  use ```@pytest.fixture``` for common setup. Moreover -

1. Add some more testcases for edge cases - like with negative values, empty strings or values outside the boundary.
2. Write tests for past deadline scenarios.
3. Add test cases for duplicate and non-existent data
4. Consider empty collection operations

**Comment by Rathore-tech-code** [comment]
Added requested tests (None title + past deadline).  
All 22 tests passing successfully.  
Kindly review again .

**Review by swaparup36 — APPROVED** [review]

File: tests/test_student_manager.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    def test_add_assignment_with_past_deadline(self, student_manager):
+        """Assignments in the past should still be tracked."""
+        past_deadline = datetime.now() - timedelta(days=1)
+
+        assignment = student_manager.add_assignment(
+            "Old HW", past_deadline, "History"
+        )
+
+        assert assignment in student_manager.assignments
```
Comment (xenonnn4w) [inline tests/test_student_manager.py:50]: why assert?
